app/access_api.py

```diff
@@ -34,6 +35,17 @@ async def abilities(self, request):
         search = dict(access=tuple(await self.auth_svc.get_permissions(request)))
         data = dict(await request.json())
         agent = (await self.data_svc.locate('agents', dict(paw=data['paw'])))[0]
-        abilities_by_executor = [await self.data_svc.locate('abilities', dict(executor=ex)) for ex in agent.executors]
-        capable_abilities = await agent.capabilities(list(itertools.chain.from_iterable(abilities_by_executor)))
-        return web.json_response([x.display for x in capable_abilities if x.access in search['access']])
+        abilities = await self.data_svc.locate('abilities', match=search)
+        capable_abilities = await agent.capabilities(list(itertools.chain.from_iterable(abilities)))
+        return web.json_response([a.display for a in capable_abilities])
+
+    async def executor(self, request):
+        data = dict(await request.json())
```

Comment:
Why do we wrap `request.json()` in a `dict()`?

Reply:
🤷 

I only see one place where this isn't the case: https://github.com/mitre/training/blob/master/app/training_api.py#L89

Reply:
Hm, i'd get rid of it if we're back in this module but this is no way a blocking request, heh